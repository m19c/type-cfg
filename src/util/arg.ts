import { TypeFunction } from '../decorators/types';

export function getTypeDecoratorArguments<T extends object>() {}

export interface ResolvedTypeDecoratorParams<T> {
  options: Partial<T>;
  typeFunction?: TypeFunction;
}
export function resolveTypeDecoratorArguments<T extends object>(
  returnTypeFuncOrOptions: TypeFunction | T | undefined,
  maybeOptions: T | undefined
): ResolvedTypeDecoratorParams<T> {
  if (typeof returnTypeFuncOrOptions === 'function') {
    return {
      typeFunction: returnTypeFuncOrOptions as TypeFunction,
      options: maybeOptions || {},
    };
  }

  return {
    options: returnTypeFuncOrOptions || {},
  };
}
