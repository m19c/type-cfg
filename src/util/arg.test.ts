import * as arg from './arg';

describe('util/arg', () => {
  describe('resolveTypeDecoratorArguments', () => {
    test('returns an object with options if the first argument is not a function', () => {
      expect(arg.resolveTypeDecoratorArguments({}, undefined)).toMatchSnapshot();
    });

    test('returns the typeFunction as well as the options', () => {
      expect(arg.resolveTypeDecoratorArguments(() => String, {})).toMatchSnapshot();
    });
  });
});
