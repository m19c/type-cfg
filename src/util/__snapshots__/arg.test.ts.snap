// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util/arg resolveTypeDecoratorArguments returns an empty options object 1`] = `
Object {
  "options": Object {},
  "typeFunction": [Function],
}
`;

exports[`util/arg resolveTypeDecoratorArguments returns an object with options if the first argument is not a function 1`] = `
Object {
  "options": Object {},
}
`;

exports[`util/arg resolveTypeDecoratorArguments returns the typeFunction as well as the options 1`] = `
Object {
  "options": Object {},
  "typeFunction": [Function],
}
`;
